### DISCUSSION TITLE
UX improvements for main workflows page

### DISCUSSION THREAD

**Discussion by EvHaus** [community/community#10060] [Actions]
The main `https://github.com/<org>/<repo>/actions` workflows UI page is not very helpful and leaves much to be desired in terms of usability and UX. I'm talking about this view:

<img width="1266" alt="Screen Shot 2022-01-12 at 3 34 16 PM" src="https://user-images.githubusercontent.com/226640/149239818-a2527af0-767d-4fd7-8511-5a05d9f2e183.png">

I have a few suggestions on how to make it 10x better.

## 1. Replace "list of workflow runs" with high level stats of workflows

The bulk of this view is occupied by a list of all runs across workflows. In a fairly large project this list is completely useless. You just see hundreds of runs in this lister across a mixture of workflows, branches and events. It's a total mess and not very useful. I know you can filter this list, but you can already filter the list by selecting a specific workflow from the left sidebar. I think a much better use of this space would be a summary report on the state of your workflows. As a user, the main things I want to know are:

- Is the main/default branch of my repo passing CI checks?
- What is the failure rate of my default branch?
- What is the speed of runs for my default branch?

Perhaps you can use Buildkite for inspiration here. Their landing page is WAY more helpful to give you an at-a-glance view on the health of your workflows (they call them pipelines):

![buildkite](https://user-images.githubusercontent.com/226640/149240228-bf620185-9cea-453b-afa4-f66efef7f16c.png)

This gives me an immediate indication on the health of my project and even has some historical context to see how things are changing over time.

## 2. Workflow run lists should be filtered by default

This is a bit subjective, but I think any time you're looking at a list of workflow runs, it should default to the a "Branch" filter re-enabled (for whatever the default branch is on your repo). Seeing a list of workflow runs across all events and branches is very noisy. It's like drinking from a firehose. I think a sensible default is to filter to only the default branch by default. This would also mirror the way `git` itself works. When you run `git log` it doesn't output a list of all commits across all your branches. That would be crazy. You just see the latest commits for your current branch. I think for CI/Actions this makes sense too.

Even more subjective, I think it should also default to having some kind of "Event" filter set as well. It's very common to have both "pull_request" and "push" events used at the same time, but it's rare that you want to see a list of them both. I think it would make sense to default to only showing "push" events so you can just look at a list of the latest runs for your default branch.

As a user, I want to see the status of the CI checks for the latest commits. And if that's too granular or I need something else I can then go and manually tweak the filters. As it is today, I've had to rely on URL bookmarks to ensure I always visit GitHub actions with `?query=event%3Apush+branch%3Amain` in the URL. I know some Actions have nothing to do with `push` events but I would place a bet that most users that use GitHub actions for their CI system do something similar. Perhaps there could be a setting to make this a default view.

## 3. Need an ability to remove old/stale workflows

On the left sidebar, under "Workflows", you see flat list of all workflows you've ever created. Including random test workflows your team may have had in the past. They show up with this icon <img width="34" alt="Screen Shot 2022-01-12 at 3 31 57 PM" src="https://user-images.githubusercontent.com/226640/149239633-6937d70b-32d2-4ef2-be48-1960666fb606.png"> but there is no way remove them from view entirely. The only way is to manually delete all runs for that workflow. This is super tedious as there is no bulk delete. In our case we have some test workflows we created with thousands of runs. I'm not going to go through them clicking buttons manually. We need a way to remove stale workflows from this list.

Ideally, GitHub would just do this automatically. Only workflows with actively configured `.github/workflows` files should show up in this list. Perhaps there can be an "archived" tab here to see everything you had in the past? As it is today, for us this list is full of garbage which we have no easy way of cleaning up.

**Comment by solita-juhosassali** [comment]
It would also be really nice to be able to filter workflows by commit SHA or similar commit related metadata. There could be multiple workflow runs with same commit SHA and currently it's difficult to find all workflow runs related to that commit.

**Comment by ethomson** [comment]
Thanks @EvHaus for the very detailed feedback.  We really appreciate it.  I like your idea of providing more details on the Actions landing page - I'll share this with the other PMs and designers.

**Comment by EvHaus** [comment]
I just came across [Trunk CI Analytics](https://trunk.io/products/ci-analytics). This is almost exactly that sort of thing I was imaging GitHub would have natively. You guys should buy them :)

<img width="659" alt="Screenshot 2023-03-30 at 2 55 00 PM" src="https://user-images.githubusercontent.com/226640/228972982-25745933-4a8b-4699-9943-54c8c32d821e.png">
